### PR TITLE
AP_Motors: compensate for attitude in tailsitter (WIP)

### PR DESCRIPTION
**Pull request by tridge**
this is a patch from Leonard to improve the tailsitter AP_Motors backend. It is a work in progress, and has yet to be flight tested

### CODE REVIEW

**Comment by tridge** [comment]
note that we need to think about how this interacts with the AC_AttitudeControl angle boost code

**Comment by kd0aij** [comment]
@tridge @lthall Flight test of the Stryker with this PR: https://drive.google.com/open?id=0Bw3digSMQXDuWEFqcWc3TzBKYkk
It flew, but there is a net increase obvious in pitch rate gain; continuous 4Hz oscillation on the P and D terms of PIQP (using the same parameters as in https://drive.google.com/open?id=0Bw3digSMQXDuTHhadjZOX2JwZzQ). 

Nothing bad happened at a pitch angle of -45 degrees though.

**Comment by rmackay9** [comment]
this patch certainly looks like an improvement.  Re angle-boost, this can be turned off by setting ATC_ANGLE_BOOST to zero (as you probably already know).

**Comment by kd0aij** [comment]
@rmackay9 thanks, I didn't know that. Will do a field test with ATC_ANGLE_BOOST=0

File: libraries/AP_Motors/AP_MotorsTailsitter.cpp [hunk 1]
```diff
@@ -101,10 +101,55 @@ void AP_MotorsTailsitter::output_to_motors()
...
+    float   throttle_thrust;            // throttle thrust input value, 0.0 - 1.0
+    float   rpy_thrust;                 // the minimum throttle setting that will not limit the roll pitch and yaw output
+    float   thr_adj;                    // the difference between the pilot's desired throttle and throttle_thrust_best_rpy
+
+    // apply voltage and air pressure compensation
+    // roll, pitch, yaw should only be scaled by air density
+    roll_thrust = _roll_in * get_compensation_gain();
+    pitch_thrust = _pitch_in * get_compensation_gain();
+    yaw_thrust = _yaw_in * get_compensation_gain();
+    throttle_thrust = get_throttle() * get_compensation_gain();
```
Comment (kd0aij) [inline libraries/AP_Motors/AP_MotorsTailsitter.cpp]: In my testing, get_compensation_gain() always returns 1.0. Is this functional for quadplanes?

**Comment by davidbuzz** [comment]
can we close this..?

**Pull request closed without merging**